q.js

```diff
@@ -1540,6 +1540,82 @@ Promise.prototype.all = function () {
 };
 
 /**
+ * Returns the first resolved promise of an array. Prior rejected promises are
+ * ignored.  Rejects only if all promises are rejected with an array
+ * containing the reasons of all rejected promises.
+ * @param {Array*} an array containing values or promises for values
+ * @returns a promise fulfilled with the value of the first resolved promise,
+ * or a promise rejected with an array with the reasons of all rejected
+ * promises.
+ */
+Q.firstResolved = firstResolved;
+
+function firstResolved(promises) {
+    var deferred = Q.defer();
+    if (promises.length === 0) {
+        deferred.resolve();
+    }
+
+    for (var i = 0; i < promises.length; i++) {
+        var promise = promises[i];
+        /* Let's define that false could be a value to resolve with, so let's not
+         * if(!promise)
+         */
+        if (promise === undefined || promise === null) {
+            continue;
+        }
+
+        if (!isPromiseAlike(promise)) {
+            // It's a value, not a promise
+            var value = promise;
+            deferred.resolve(value);
+            return deferred.promise;
+        }
```

Comment:
Above block is superfluous. Q.resolve(promise) will coerce promises, promise-alikes, and non-promises into a promise. Apply liberally to all arguments and consistency is solved.